spring-boot/autoconfigure/src/main/java/org/togglz/spring/boot/autoconfigure/TogglzAutoConfiguration.java

```diff
@@ -0,0 +1,251 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.togglz.core.Feature;
+import org.togglz.core.activation.ActivationStrategyProvider;
+import org.togglz.core.activation.DefaultActivationStrategyProvider;
+import org.togglz.core.manager.CompositeFeatureProvider;
+import org.togglz.core.manager.EnumBasedFeatureProvider;
+import org.togglz.core.manager.FeatureManager;
+import org.togglz.core.manager.FeatureManagerBuilder;
+import org.togglz.core.manager.PropertyFeatureProvider;
+import org.togglz.core.repository.StateRepository;
+import org.togglz.core.repository.cache.CachingStateRepository;
+import org.togglz.core.repository.composite.CompositeStateRepository;
+import org.togglz.core.repository.file.FileBasedStateRepository;
+import org.togglz.core.repository.mem.InMemoryStateRepository;
+import org.togglz.core.spi.ActivationStrategy;
+import org.togglz.core.spi.FeatureProvider;
+import org.togglz.core.user.NoOpUserProvider;
+import org.togglz.core.user.UserProvider;
+import org.togglz.spring.boot.autoconfigure.TogglzProperties.FeatureSpec;
+import org.togglz.spring.listener.TogglzApplicationContextBinderApplicationListener;
+import org.togglz.spring.security.SpringSecurityUserProvider;
+import org.togglz.spring.web.FeatureInterceptor;
+
+import com.github.heneke.thymeleaf.togglz.TogglzDialect;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for Togglz.
+ *
+ * @author Marcel Overdijk
+ */
+@Configuration
+@ConditionalOnProperty(prefix = "togglz", name = "enabled", matchIfMissing = true)
+@EnableConfigurationProperties(TogglzProperties.class)
+public class TogglzAutoConfiguration {
+
+    @Bean
+    public TogglzApplicationContextBinderApplicationListener togglzApplicationContextBinderApplicationListener() {
+        return new TogglzApplicationContextBinderApplicationListener();
+    }
+
+    @Configuration
+    @ConditionalOnMissingBean(FeatureProvider.class)
+    protected static class FeatureProviderConfiguration {
```

Comment:
Why don't you move it to a separate class?

Reply:
That is a nested `@Configuration`, it's more clean this way, otherwise we would have to `@Import` it, and besides it inheriths the `@Conditional`s from `TogglzAutoConfiguration`